Integrate with HTML, part 1 of n

domenic opened the pull request:
This is the counterpart to https://github.com/whatwg/html/pull/1153, which removes the parts of the spec that are now defined in HTML: the content and IDL attributes on various elements, plus the `<meta http-equiv>` extension.

There is still much to do; I will open a separate issue detailing the plan.

jeisinger commented:
Can you please update the "Integration with HTML section" to point out the changes that need to be done to the W3C HTML spec (and preferably file spec bugs for them), similar to what we have in CSP: https://w3c.github.io/webappsec-csp/#html-integration

domenic commented:
Can do, although I don't think that's really necessary...

jeisinger commented:
for better or worse, this spec should make sense in the W3C context as well

domenic commented:
Done

Ms2ger commented:
I disagree that there's any point to mentioning the harmful w3c fork.

jeisinger commented:
@Ms2ger please bring this up with Wendy, who's working on this issue

mikewest commented:
Two broad points:

1. I think it is good for a document describing a single feature to describe all of it in a way that web developers and browser implementers can skim in one sitting, rather than hopping back and forth between HTML, DOM, Fetch, and etc. Those are and should be the normative references, but this document should have enough on its bones to be a cohesive picture of referrer policy. Dropping the integration text entirely seems counterproductive. I think CSP has a reasonable compromise at the moment, and the effort required to maintain that section is IMO outweighed by its value to the reader.

2. I agree that the split between whatwg/html and w3c/html is frustrating, annoying, duplicitous, error-prone, and likely to result in developer confusion. As long as we're in a world in which folks are working on both, I think it's reasonable for us to try to upstream to both. That said, I personally am not spending time either reviewing or creating PRs for w3c/html, as it's not the document I use when implementing. I do file bugs against that spec, and I think it's worth doing so, if only to make clear the amount of effort involved in maintaining and updating the document.

One narrower point: as long as we're driving this spec through the WebAppSec WG, the snapshots we produce will need to reference documents that body considers stable. It's totally fine for the ED to reference WHATWG docs; @wseltzer, @dveditz, and @hillbrad are going to need to work out how to get those documents past WD to REC.

For better or worse, this is a problem that we've been kicking down the field starting with the CSP2 last call ~2 years ago. We've got a face-to-face meeting in Mountain View on the 16th/17th. I hope we can make some progress there.

domenic commented:
Can you expand on what your (1) means in practice for this PR? Are you saying that the content and IDL attributes should not move to HTML?

mikewest commented:
Naah. They should absolutely move to HTML. For this document, I'd suggest that we mention them, explain their purpose, and etc, in a non-normative "Integration with HTML" section, similar conceptually to what the CSP draft is doing with various HTML concepts.

I'd like folks to be able to read _this_ document, and understand everything they need to understand. Non-normative duplication that points out to an official source seems like the right way to handle that.

domenic commented:
That makes sense. I'd ideally like to leave that to the authors of this document, since they're in a better position to do so; I included a couple sentences as a pointer, which was about as much more information as existed in the spec previously, but I think you are imaging more of a lengthy tutorial.

I am still hoping this can be merged soon-ish so we can continue with the HTML integration plan described in #43.

domenic commented:
Although I guess this should not be merged until https://github.com/whatwg/html/pull/1153 is reviewed.

estark37 commented:
This PR lgtm. Once it's merged I can take a look through and see if we need a separate HTML integration section.

estark37 commented:
Forgot to say, thanks for doing this work @domenic! And sorry I'm sooooooo slow reviewing.

tabatkins commented:
> One narrower point: as long as we're driving this spec through the WebAppSec WG, the snapshots we produce will need to reference documents that body considers stable. It's totally fine for the ED to reference WHATWG docs; @wseltzer, @dveditz, and @hillbrad are going to need to work out how to get those documents past WD to REC

There is absolutely nothing wrong with reffing WHATWG specs in later W3C stability levels. You just need to show that they are reasonably stable, like reffing any other document outside of W3C.

mikewest commented:
@tabatkins: I agree with you, so I'll rephrase for clarity: so far I have not been successful when attempting to documents that reference WHATWG's HTML or Fetch to CR+.

domenic commented:
OK, @estark37 I updated this to also include the switch to string-named referrer policies. It seems like a really nice improvement, upon viewing the results. The document was rather inconsistent in how it referred to the policies, and this totally clarifies the empty string business.

Would love to get this merged!

estark37 commented:
Thanks! I'll go ahead and merge this but @jeisinger I think it would still be good for you take a look when you have a chance.

domenic commented:
\o/! I'll get HTML updated posthaste.

domenic commented:
Oh, how do we update gh-pages? It looks like it is behind master by a decent bit, not just these commits.

estark37 commented:
@domenic yeah, I noticed that the other day... I'm not sure how we do that. I suppose I'll just trying resetting it to master and push -f and see what happens. Wish me luck

domenic commented:
I think `make publish` maybe? But on the computer of someone with access to the repo.

estark37 commented:
Oh, cool, that did it. I guess I have access to the repo. :)

jdm commented:
Can we update the referrer-policy tests in https://github.com/w3c/web-platform-tests/ as well?

domenic commented:
What is there to update as a result of this PR?

jdm commented:
The change from `origin-only` to `origin`?

domenic commented:
The string did not change, only the spec section title





On Tue, May 24, 2016 at 1:56 PM -0700, "Josh Matthews" <notifications@github.com<mailto:notifications@github.com>> wrote:


The change from origin-only to origin?

-
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/w3c/webappsec-referrer-policy/pull/42#issuecomment-221388814>

jdm commented:
@domenic ae3e8933125caa5f39eeadff235eeee49ac12ee1 appears to disagree with you? Or maybe a16588962af5c59e8c056fb852f98293c00b5862 is more relevant, but it looks like the string was changed without updating the tests.

domenic commented:
The string was always "origin". The state was name "Origin Only", which caused confusion. When I removed the state name vs. string distinction, I accidentally made a normative change to change the string from "origin" to "origin-only". This was accidental, so I reverted it in the commits you cite.

domenic commented:
You can see the old spec at https://rawgit.com/w3c/webappsec-referrer-policy/5874e73ddf780ca90a4e0b206df3194c393648e3/index.html to confirm that the string has always been "origin".

jdm commented:
I see. Thanks for the clarification!
